parity-bytes/src/lib.rs

```diff
@@ -19,9 +19,27 @@
 //! Includes a pretty-printer for bytes, in the form of `ToPretty` and `PrettySlice`
 //! as
 
-use std::fmt;
-use std::cmp::min;
-use std::ops::{Deref, DerefMut};
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+// Re-export libcore using an alias so that the macros can work without
+// requiring `extern crate core` downstream.
```

Comment:
Not sure I understand what you mean by "the macros" – are there macros exported from this crate?

Reply:
to be fair it seems to be copy-pasted across many crates: https://github.com/search?l=Rust&q=Re-export+libcore+using+an+alias+so+that+the+macros+can+work+without&type=Code

Reply:
I get it from the `elastic-array` and it have some marcos exported.
https://github.com/debris/elastic-array/blob/8f01572f850e6f387ad785fb1e59d599eb260ea0/src/lib.rs#L11-L28

I'm not sure whether  will be some macros exported from this crate, so I didn't delete it :smile:

Reply:
So if we’re not doing it for the macros, then what is the purpose of this re-export? Can we remove it?

Reply:
removed. :ok_hand:

Reply:
I delete the re-export,it show the following errors:
https://travis-ci.org/paritytech/parity-common/jobs/533187061#L1447
https://ci.appveyor.com/project/paritytech/parity-common/builds/24584892

Reply:
So I think you need to do:

```
#[cfg(feature = "std")]
extern crate core;
```
(notice no `pub`)

…and then replace `use core_::{… …` with `use core::{… …`.

The idea is that when running with `std` you need to import the `core` as a crate. In `no_std` mode `core` is part of the prelude.

Reply:
Thanks for your guide!